pkg/controllers/uiplugin/monitoring.go

```diff
@@ -0,0 +1,123 @@
+package uiplugin
+
+import (
+	"fmt"
+	"strings"
+
+	osv1 "github.com/openshift/api/console/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	uiv1alpha1 "github.com/rhobs/observability-operator/pkg/apis/uiplugin/v1alpha1"
+)
+
+func createMonitoringPluginInfo(plugin *uiv1alpha1.UIPlugin, namespace, name, image string, features []string) (*UIPluginInfo, error) {
+	config := plugin.Spec.Monitoring
+	if config == nil {
+		return nil, fmt.Errorf("monitoring configuration can not be empty for plugin type %s", plugin.Spec.Type)
+	}
+
+	if config.Alertmanager.Url == "" {
+		return nil, fmt.Errorf("alertmanager location can not be empty for plugin type %s", plugin.Spec.Type)
+	}
+	if config.ThanosQuerier.Url == "" {
+		return nil, fmt.Errorf("ThanosQuerier location can not be empty for plugin type %s", plugin.Spec.Type)
+	}
+
+	pluginInfo := &UIPluginInfo{
+		Image:       image,
+		Name:        name,
+		ConsoleName: "monitoring-console-plugin",
+		DisplayName: "Monitoring Console Plugin",
+		ExtraArgs: []string{
+			fmt.Sprintf("-features=%s", strings.Join(features, ",")),
+			fmt.Sprintf("-alertmanager=%s", config.Alertmanager.Url),
+			fmt.Sprintf("-thanos-querier=%s", config.ThanosQuerier.Url),
+		},
+		ResourceNamespace: namespace,
+		Proxies: []osv1.ConsolePluginProxy{
+			{
+				Alias:         "backend",
+				Authorization: "UserToken",
+				Endpoint: osv1.ConsolePluginProxyEndpoint{
+					Type: osv1.ProxyTypeService,
+					Service: &osv1.ConsolePluginProxyServiceConfig{
+						Name:      name,
+						Namespace: namespace,
+						Port:      port,
+					},
+				},
+			},
+			{
+				Alias:         "alertmanager-proxy",
+				Authorization: "UserToken",
+				Endpoint: osv1.ConsolePluginProxyEndpoint{
+					Type: osv1.ProxyTypeService,
+					Service: &osv1.ConsolePluginProxyServiceConfig{
+						Name:      name,
+						Namespace: namespace,
+						Port:      9444,
+					},
+				},
+			},
+			{
+				Alias:         "thanos-proxy",
+				Authorization: "UserToken",
+				Endpoint: osv1.ConsolePluginProxyEndpoint{
+					Type: osv1.ProxyTypeService,
+					Service: &osv1.ConsolePluginProxyServiceConfig{
+						Name:      name,
+						Namespace: namespace,
+						Port:      9445,
+					},
+				},
+			},
```

Comment:
Do we need this independent proxies or could we use only the `backend` adding a `proxy` endpoint?, e.g.
`backend/proxy/alertmanager/...` this would mean we have control of the proxies from the plugin side.

Reply:
Unfortunately, the "backend" parameter only works for GET requests, a "proxy" is needed for every other type. The alert-manager proxy uses DELETE for silences, and while I'm not sure if the thanos-proxy needs any other request types, I left it like this for safety